File: app/src/main/java/com/arbaelbarca/listmovieskatalog/adapter/AdapterTagsMovies.java
package com.arbaelbarca.listmovieskatalog.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.arbaelbarca.listmovieskatalog.R;
import com.arbaelbarca.listmovieskatalog.model.modeldetailmovies.GenresItem;

import java.util.ArrayList;

public class AdapterTagsMovies extends RecyclerView.Adapter<AdapterTagsMovies.ViewHolder> {
    private Context context;
    private ArrayList<GenresItem> genresItems;

    public AdapterTagsMovies(Context context) {
        this.context = context;
        genresItems = new ArrayList<>();
    }

    public void setGenresItems(ArrayList<GenresItem> genresItems) {
        this.genresItems = genresItems;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context)
                .inflate(R.layout.layout_item_tags, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        GenresItem genresItem = genresItems.get(position);
        holder.txtTag.setText(genresItem.getName());
    }

    @Override
    public int getItemCount() {
        return genresItems.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView txtTag;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            txtTag = itemView.findViewById(R.id.tv_tag);
        }
    }
}

File: app/src/main/java/com/robytakwa/listpicturekatalog/adapter/AdapterTagsMovies.kt
package com.robytakwa.listpicturekatalog.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.robytakwa.listpicturekatalog.R
import com.robytakwa.listpicturekatalog.model.modeldetailmovies.GenresItem
import java.util.*

class AdapterTagsMovies(private val context: Context) : RecyclerView.Adapter<AdapterTagsMovies.ViewHolder>() {
    private var genresItems: ArrayList<GenresItem>? = null

    init {
        genresItems = ArrayList()
    }

    fun setGenresItems(genresItems: ArrayList<GenresItem>) {
        this.genresItems = genresItems
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context)
                .inflate(R.layout.layout_item_tags, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        val genresItem = genresItems!![position]
        holder.txtTag.text = genresItem.name
    }

    override fun getItemCount(): Int {
        return genresItems!!.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var txtTag: TextView = itemView.findViewById(R.id.tv_tag)

    }
}
